### ISSUE TITLE
Using TensorFlow to Train a Shallow NN with Stochastic Gradient Descent

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#12112]
Using TensorFlow to Train a Shallow NN with Stochastic Gradient Descent<br>
<br>
https://ift.tt/HCFy7XS<br>
<br>
Lee Meng<br>
<br>

<div><div><div><div>
<div>
<p>The goal here is to progressively train deeper and more accurate models using TensorFlow. We will first load the notMNIST dataset which we have done data cleaning. For the classification problem, we will first train two logistic regression models use simple gradient descent, stochastic gradient descent (SGD) respectively for optimization to see the difference between these optimizers.</p>
<p>Finally, train a Neural Network with one-hidden layer using ReLU activation units to see whether we can boost our model's performance further.</p>
<p>Previously in <a href="https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/udacity/1_notmnist.ipynb">1_notmnist.ipynb</a>, we created a pickle with formatted datasets for training, development and testing on the <a href="http://yaroslavvb.blogspot.com/2011/09/notmnist-dataset.html">notMNIST dataset</a>.</p>
<p>This post is modified from the <a href="https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/udacity/2_fullyconnected.ipynb">jupyter notebook</a> originated from the Udacity MOOC course: <a href="https://www.udacity.com/course/deep-learning--ud730">Deep learning by Google</a>.</p>
</div>
</div></div></div></div><br>
<br>
<br>
<br>
<br>
<br>
via LeeMeng - All About DataScience <br>
<br>
April 21, 2025 at 10:34AM
